internal/generate/endpoints/main.go

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+//go:build generate
+// +build generate
+
+package main
+
+import (
+	_ "embed"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"html/template"
+	"io"
+	"net/http"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/aws-sdk-go-base/v2/internal/generate/common"
+	"github.com/hashicorp/aws-sdk-go-base/v2/internal/slices"
+)
+
+type PartitionDatum struct {
+	ID          string
+	Name        string
+	DNSSuffix   string
+	RegionRegex string
+	Regions     []RegionDatum
+}
+
+type RegionDatum struct {
+	ID          string
+	Description string
+}
+
+type TemplateData struct {
+	Partitions []PartitionDatum
+}
+
+func usage() {
+	fmt.Fprintf(os.Stderr, "Usage:\n")
+	fmt.Fprintf(os.Stderr, "\tmain.go <aws-sdk-go-v2-endpoints-json-url>\n\n")
+}
+
+func main() {
+	flag.Usage = usage
+	flag.Parse()
+
+	args := flag.Args()
+
+	if len(args) < 1 {
+		flag.Usage()
+		os.Exit(2)
+	}
+
+	inputURL := args[0]
```

Comment:
Using the GitHub content URL works, or you could use functions from https://pkg.go.dev/golang.org/x/mod to parse the modfile (https://pkg.go.dev/golang.org/x/mod@v0.21.0/modfile) and get the path to the files on disk (https://pkg.go.dev/golang.org/x/mod/module)

Reply:
Thanks for the suggestion -- will stick with what we have for now; we can consider switching if we find the download unreliable (it isn't run as part of a release and is likely to be run by a maintainer).